hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/map/AbstractMapPutMessageTask.java

```diff
@@ -36,17 +38,17 @@ protected AbstractMapPutMessageTask(ClientMessage clientMessage, Node node, Conn
 
     @Override
     protected void beforeProcess() {
-        startTime = System.currentTimeMillis();
+        startTime = System.nanoTime();
     }
 
     @Override
     protected void beforeResponse() {
-        final long latency = System.currentTimeMillis() - startTime;
+        final long latency = System.nanoTime() - startTime;
```

Comment:
To prevent confusion; I always add the unit of time to to the name of the variable. So latencyNanos. Dealing with unit mismatch is a PITA.